src/validation_workflow/docker/validation_docker.py

```diff
@@ -0,0 +1,203 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import shutil
+import subprocess
+import time
+from subprocess import PIPE
+from typing import Any
+
+import requests
+
+from system.temporary_directory import TemporaryDirectory
+from validation_workflow.api_request import ApiTest
+from validation_workflow.api_test_cases import TestCases
+from validation_workflow.docker.inspect_docker_image import InspectDockerImage
+from validation_workflow.validation import Validation
+from validation_workflow.validation_args import ValidationArgs
+
+
+class ValidateDocker(Validation):
+
+    def __init__(self, args: ValidationArgs) -> None:
+        super().__init__(args)
+
+    def download_artifacts(self) -> bool:
+
+        # Check if the docker daemon is running; or else, bring it up
+        if (self.is_container_daemon_running() == 0):
+            logging.info('docker_daemon is running')
+
+            # STEP 1 . pull the images for OS and OSD
+            _OS_image_name = 'opensearchproject/opensearch' if self.args.dockersource == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
+            _OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.dockersource == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+
+            self.local_image_OS_id = self.get_image_id(_OS_image_name, self.args.version)
+            self.local_image_OSD_id = self.get_image_id(_OSD_image_name, self.args.version)
+            logging.info('the OS image ID is : ' + self.local_image_OS_id)
+            logging.info('the OSD image ID is : ' + self.local_image_OSD_id + '\n\n')
+
+            return True
+
+        else:
+            raise Exception('docker_daemon check is fail. Exiting the validation.')
+
+    def installation(self) -> bool:
+        ''' Pass this method for docker as no installation required in docker'''
+        return True
+
+    def start_cluster(self) -> bool:
+        ''' Pass this method for docker and combine it with the following method because we want to Pass the digest validation in docker first before spinning up the docker container'''
```

Comment:
Same as above

Reply:
Thanks, @gaiksaya , sounds good, let us update it at next commit.